src/Aspire.Dashboard/Components/Pages/Traces.razor.cs

```diff
@@ -267,21 +272,88 @@ public void UpdateViewModelFromQuery(TracesPageViewModel viewModel)
     {
         viewModel.SelectedApplication = _applicationViewModels.GetApplication(Logger, ApplicationName, canSelectGrouping: true, _allApplication);
         TracesViewModel.ApplicationKey = PageViewModel.SelectedApplication.Id?.GetApplicationKey();
+
+        if (SerializedLogFilters is not null)
+        {
+            var filters = LogFilterFormatter.DeserializeLogFiltersFromString(SerializedLogFilters);
+
+            if (filters.Count > 0)
+            {
+                TracesViewModel.ClearFilters();
+                foreach (var filter in filters)
+                {
+                    TracesViewModel.AddFilter(filter);
+                }
+            }
+        }
+
+        _ = Task.Run(async () =>
```

Comment:
should we make the parent method async?

Reply:
I copied this from some other UI that did the same thing. I'll come back and look at improving both places later.